internal/controller/apiextensions/composite/api.go

```diff
@@ -198,27 +197,42 @@ func (f *APIRevisionFetcher) Fetch(ctx context.Context, cr resource.Composite) (
 	return AsComposition(current), nil
 }
 
-// currentRevision returns the current revision of the supplied composition. It
-// returns nil if none of the supplied revisions appear to be currentRevision.
+func (f *APIRevisionFetcher) getCompositionRevisionList(ctx context.Context, cr resource.Composite, comp *v1.Composition) (*v1alpha1.CompositionRevisionList, error) {
+	rl := &v1alpha1.CompositionRevisionList{}
+	ml := client.MatchingLabels{}
+
+	if cr.GetCompositionUpdatePolicy() != nil && *cr.GetCompositionUpdatePolicy() == xpv1.UpdateAutomatic &&
+		cr.GetCompositionRevisionSelector() != nil && len(cr.GetCompositionRevisionSelector().MatchLabels) != 0 {
+		ml = cr.GetCompositionRevisionSelector().MatchLabels
+	}
```

Comment:
```suggestion
	if cr.GetCompositionUpdatePolicy() != nil && *cr.GetCompositionUpdatePolicy() == xpv1.UpdateAutomatic && cr.GetCompositionRevisionSelector() != nil {
		ml = cr.GetCompositionRevisionSelector().MatchLabels
	}
```

Nit: I think this would have the same result, with fewer conditionals. If `cr.GetCompositionRevisionSelector().MatchLabels` is a zero length array there's no harm in setting `ml = cr.GetCompositionRevisionSelector().MatchLabels` (which was already a zero length array).